R/misc.R

```diff
@@ -38,12 +38,15 @@ add_class <- function(x, cls) {
   x
 }
 
-strata_check <- function(strata, vars) {
+strata_check <- function(strata, data) {
   if (!is.null(strata)) {
     if (!is.character(strata) | length(strata) != 1) {
       rlang::abort("`strata` should be a single character value.")
     }
-    if (!(strata %in% vars)) {
+    if (inherits(data[, strata], "Surv")) {
+      rlang::abort("`strata` cannot be a `Surv` object. Use the time or event indicator directly.")
```

Comment:
What do you think about:
```suggestion
      rlang::abort("`strata` cannot be a `Surv` object. Use the time or event variable directly.")
```

Reply:
sounds good :raised_hands: